pkg/store/bucket.go

```diff
@@ -1240,7 +1253,9 @@ func debugFoundBlockSetOverview(logger log.Logger, mint, maxt, maxResolutionMill
 }
 
 // Series implements the storepb.StoreServer interface.
-func (s *BucketStore) Series(req *storepb.SeriesRequest, srv storepb.Store_SeriesServer) (err error) {
+func (s *BucketStore) Series(req *storepb.SeriesRequest, seriesSrv storepb.Store_SeriesServer) (err error) {
+	srv := newFlushableServer(seriesSrv, s.LabelNamesSet(), req.WithoutReplicaLabels)
```

Comment:
This will be a yet another way of performing a race inside of the storegateway but let's work on fixing this now before the release :+1: :smile:

Reply:
I actually might be wrong because the Close() functions will still run at the same time :thinking: let's merge this and test it